qiskit/circuit/bit.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Quantum bit and Classical bit objects.
+"""
+
+from qiskit.exceptions import QiskitError
+
+class Bit:
+    """Implement a generic bit."""
+
+    def __init__(self, register, index):
+        """Create a new generic bit.
+        """
+        try:
+            size = int(index)
+        except Exception:
+            raise QiskitError("index needs to be castable to an int")
+        if size < 0:
+            raise QiskitError("index size must be positive")
+
+        self.register = register
+        self.index = index
+
+    def __repr__(self):
+        """Return the official string representing the bit."""
+        return "%s(%s, %s)" % (self.__class__.__name__, self.register, self.index)
+
+    def __lt__(self, other):
+        if self.register == other.register:
+            return self.index < other.index
+        return self.register < other.register
```

Comment:
Hi! Where is `Register < Register` defined?

Reply:
Good catch! fixed in d042a1eb

Reply:
Do we need to define an ordering for `Bit`s and `Register`s, or is there a use case that requires them? `Bit`s are naturally ordered within a `Register` (but only in the context of that register). The order we define here generally won't be the same as that returned from e.g. `quantumcircuit.qubits` and it seems like that could be a likely source of confusion.

Reply:
Many tests are doing `sorted()` to qargs. I assumed it is because python 3.5/3.6 testing... It's the way that it is code now. We can open an issue about it...